draft-ietf-masque-connect-udp-listen.md

```diff
@@ -172,7 +181,18 @@ and UDP Port specified in each Listener Datagram Payload received from the
 client. The proxy uses the same port to listen for UDP packets from any
 authorized target and encapsulates the packets in the Listener Datagram
 Payload format, specifying the IP and port of the target and forwards it to
-the client.
+the client. The proxy MUST keep a mapping of Context-IDs to IP Addresses and UDP
+Ports. If the client sends a payload defined in {#format} with the IP Version,
+IP Address and UDP Port fields, the proxy MUST update its mapping of the context
+ID to the provided IP Address and UDP Port. If the client sends a datagram
+without the IP Version, IP Address and UDP Port fields, the proxy MUST use the
```

Comment:
How does the receiver know if the version/address/port is included or omitted?

Reply:
Yeah, I don't see how either side could know which to expect. I would think we'd need to change context ID, not just use the same one?